client/store/current-user/test/selectors.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import selectors from '../selectors';
+import { select } from '@wordpress/data';
+
+const { getCurrentUser, isRequestingCurrentUser } = selectors;
+jest.mock( '@wordpress/data', () => ( {
+	...require.requireActual( '@wordpress/data' ),
+	select: jest.fn().mockReturnValue( {} ),
+} ) );
+
+describe( 'getCurrentUser()', () => {
+	it( 'returns null when no user data is available', () => {
+		const state = deepFreeze( {} );
+		expect( getCurrentUser( state ) ).toEqual( null );
+	} );
+	it( 'returns current user information', () => {
+		let state = deepFreeze( {
+			currentUser: {
+				data: {},
+			},
+		} );
+		expect( getCurrentUser( state ) ).toEqual( {} );
+
+		state = deepFreeze( {
+			currentUser: {
+				data: {
+					first_name: 'Testing',
+				},
+			},
+		} );
+		expect( getCurrentUser( state ) ).toEqual( {
+			first_name: 'Testing',
+		} );
```

Comment:
I'm getting an error with this one, although the tests pass

<img width="668" alt="screen shot 2018-08-06 at 12 02 10 pm" src="https://user-images.githubusercontent.com/1922453/43691442-bff7db3c-9970-11e8-8757-acf5faafc002.png">